media-gfx/blender/blender-9999.ebuild

```diff
@@ -36,7 +35,7 @@ SLOT="${PV%.*}"
 IUSE="
 	alembic +bullet collada +color-management cuda +cycles +cycles-bin-kernels
 	debug doc +embree experimental +ffmpeg +fftw +fluid +gmp gnome hip jack
-	jemalloc jpeg2k man +nanovdb ndof nls +oidn oneapi openal +openexr +openmp openpgl
+	jemalloc jpeg2k man +nanovdb ndof nls +oidn oneapi openal +openexr +openmp +openpgl
```

Comment:
Hi, just noticed this because the binhost now fails to build blender for the x86-64 baseline. The commit message doesn't really specify why it changed. "media-gfx/blender: Update live ebuild" is not a great commit message, even though I'm sure you had your reasons for changing this.

(Please be more careful about commit messages in the future!!!)

I guess mainly what I want to know is, what should users of e.g. baseline binhosts be expected to do here? Given that openpgl does not build unless you manually opt into at least one CPU_FLAGS_X86.

Reply:
Note that *all* options in CPU_FLAGS_X86 are defaulted on via profiles/arch/amd64/make.defaults, if they are available on the x86-64-v1 baseline. Anything more than the baseline requires specific CPUs that users may or may not have, which means that openpgl is effectively impossible to install for users of the first generation of amd64 CPUs.

openpgl supports (by default all disabled): `CPU_FLAGS_X86="-avx2 -avx512dq -sse4_2"`

Reply:
Ah... This is an issue. Blender now only officially supports CPUs that can do sse4.2 and up.
https://www.blender.org/download/requirements/

My intent was to enable and many defaults as possible to keep the default gentoo configuration as close to upstream as possible. (At least for non obvious dependencies like openpgl that greatly speeds up cycles rendering)
While it is still possible to strip down Blender to build and run it on non sse4.2 cpus, I think it is better to prepare ourselves for the future.

Perhaps we should add a sse4.1 requirement to Blender itself?

Reply:
> Perhaps we should add a sse4.2 requirement to Blender itself?

The only way to do that sensibly really, is a USE flag. Defaulting to `+openpgl` is effectively this anyway so it seems like a bit of a moot point. Especially if blender still *works* without sse4.2 as long as you disable USE=openpgl, "advisability of running on slow machines" or not.

Even "preparing ourselves for the future" is probably better done as a default rather than a hard cutoff.

Reply:
So at this point we don't change anything?
I'm just making sure that I didn't misunderstand anything.

Reply:
The `x86-64-v1` builders have `-openpgl` now.

I'll come up with something for the blender ebuilds to make it clearer that blender wants SSE4.2 and openpgl requires it.